packages/good-design/src/hooks/useFVModalAction.ts

```diff
@@ -43,8 +43,8 @@ export const useFVModalAction = ({ firstName, method, onClose }: FVModalActionPr
 
     switch (method) {
       case "redirect": {
-        const link = fvlink?.getLink(firstName, document.location.href, false);
-        console.log("test for redirect back link -->", { link });
+        const callbackUrl = document.location.href.replace("#", "%23");
```

Comment:
1.  fvlink?.getLink should take care of this
2. there's an URL class allowing you to parse / assemble urls and working with all parts correctly including the hash

so you could
a) create URL from callbackUrl
b) add query params using .searchParams.append
c) assemble url back to string. the original hash value should be preserved

Reply:
@L03TJ3 the issue is at `fvid` side. I'll hot fix gooddapp repo, this could be reverted